server/src/main/java/org/elasticsearch/search/profile/AbstractProfileBreakdown.java

```diff
@@ -25,34 +28,46 @@ public abstract class AbstractProfileBreakdown<T extends Enum<T>> {
     /**
      * The accumulated timings for this query node
      */
-    private final Timer[] timings;
+    private final Map<Integer, List<Timer>> timings;
     private final T[] timingTypes;
 
     /** Sole constructor. */
     public AbstractProfileBreakdown(Class<T> clazz) {
         this.timingTypes = clazz.getEnumConstants();
-        timings = new Timer[timingTypes.length];
-        for (int i = 0; i < timings.length; ++i) {
-            timings[i] = new Timer();
+        timings = new HashMap<>(this.timingTypes.length);
+        for (int i = 0; i < this.timingTypes.length; ++i) {
+            ArrayList<Timer> listOfTimers = new ArrayList<>();
+            listOfTimers.add(new Timer());
+            timings.put(i, listOfTimers);
```

Comment:
as stated below, why are we keeping an array of enum values? An enum value has hash codes and equality checks so can be used as hashmap keys.